code/modules/flufftext/Hallucination.dm

```diff
@@ -19,187 +19,349 @@ mob/living/carbon/
+	processing_objects += src
+
+/obj/effect/hallucination/fake_flood/process()
+	if(next_expand <= world.time)
+		radius++
+		if(radius > FAKE_FLOOD_MAX_RADIUS)
+			qdel(src)
+		Expand()
+		next_expand = world.time + FAKE_FLOOD_EXPAND_TIME
+	return
+
+/obj/effect/hallucination/fake_flood/proc/Expand()
+	if(!flood_turfs) //for qdel
+		return
+	for(var/turf/T in circlerangeturfs(loc,radius))
+		if((T in flood_turfs)|| T.blocks_air)
+			continue
+		flood_images += image(image_icon,T,image_state,MOB_LAYER)
+		flood_turfs += T
+	if(target.client) target.client.images |= flood_images
+	return
+
+/obj/effect/hallucination/fake_flood/Destroy()
+	processing_objects -= src
+	del(flood_turfs)
+	if(target.client) target.client.images.Remove(flood_images)
+	target = null
+	del(flood_images)
+	return
+
+/obj/effect/hallucination/simple/xeno
+	image_icon = 'icons/mob/alien.dmi'
+	image_state = "alienh_pounce"
+
+/obj/effect/hallucination/simple/xeno/New(loc,var/mob/living/carbon/T)
+	..()
+	name = "alien hunter ([rand(1, 1000)])"
+	return
 
-/mob/living/carbon/proc/fake_attack()
-//	var/list/possible_clones = new/list()
+/obj/effect/hallucination/simple/xeno/throw_at(atom/target, range, speed) // TODO : Make diagonal trhow into proc/property
+	if(!target || !src || (flags & NODROP))	return 0
+
+	src.throwing = 1
+
+	var/dist_x = abs(target.x - src.x)
+	var/dist_y = abs(target.y - src.y)
+	var/dist_travelled = 0
+	var/dist_since_sleep = 0
+
+	var/tdist_x = dist_x;
+	var/tdist_y = dist_y;
+
+	if(dist_x <= dist_y)
+		tdist_x = dist_y;
+		tdist_y = dist_x;
+
+	var/error = tdist_x/2 - tdist_y
+	while(target && (((((dist_x > dist_y) && ((src.x < target.x) || (src.x > target.x))) || ((dist_x <= dist_y) && ((src.y < target.y) || (src.y > target.y))) || (src.x > target.x)) && dist_travelled < range) || !has_gravity(src)))
+
+		if(!src.throwing) break
+		if(!istype(src.loc, /turf)) break
+
+		var/atom/step = get_step(src, get_dir(src,target))
+		if(!step)
+			break
+		src.Move(step, get_dir(src, step))
+		hit_check()
+		error += (error < 0) ? tdist_x : -tdist_y;
+		dist_travelled++
+		dist_since_sleep++
+		if(dist_since_sleep >= speed)
+			dist_since_sleep = 0
+			sleep(1)
+
+
+	src.throwing = 0
+	src.throw_impact(get_turf(src))
+
+	return 1
+
+/obj/effect/hallucination/simple/xeno/throw_impact(A)
+	update_icon("alienh_pounce")
+	if(A == target)
+		target.Weaken(5)
+		target.visible_message("<span class='danger'>[target] flails around wildly.</span>","<span class ='userdanger'>[name] pounces on you!</span>")
+
+/obj/effect/hallucination/xeno_attack
+	//Xeno crawls from nearby vent,jumps at you, and goes back in
+	var/obj/machinery/atmospherics/unary/vent_pump/pump = null
+	var/obj/effect/hallucination/simple/xeno/xeno = null
+
+/obj/effect/hallucination/xeno_attack/New(loc,var/mob/living/carbon/T)
+	target = T
+	for(var/obj/machinery/atmospherics/unary/vent_pump/U in orange(7,target))
+		if(!U.welded)
+			pump = U
+			break
+	xeno = new(pump.loc,target)
```

Comment:
This will runtime if someone gets this hallucination while they aren't within 7 tiles of a vent. Also, `orange` will pick vents that aren't even in your view, and I have a feeling that always grabbing the first one found will bias it towards a certain direction.